src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -4997,7 +5488,256 @@ static void Main()
+    public static C operator+(C c, Func<object> f) { Console.WriteLine(""operator+(C c, Func<object> f)""); return c; }
+    static void Main()
+    {
+        var c = new C();
+        _ = c + (() => new object());
+        _ = c + (() => 1);
+    }
+}";
+
+            var expectedOutput =
+@"operator+(C c, Func<object> f)
+operator+(C c, Func<object> f)
+";
+            CompileAndVerify(source, parseOptions: TestOptions.Regular9, expectedOutput: expectedOutput);
+            CompileAndVerify(source, parseOptions: TestOptions.Regular10, expectedOutput: expectedOutput);
+            CompileAndVerify(source, expectedOutput: expectedOutput);
+        }
+
+        [Fact]
+        public void BinaryOperator_07()
+        {
+            var source =
+@"using System;
+using System.Linq.Expressions;
+class C
+{
+    public static C operator+(C c, Expression e) { Console.WriteLine(""operator+(C c, Expression e)""); return c; }
+    public static C operator+(C c, Func<object> f) { Console.WriteLine(""operator+(C c, Func<object> f)""); return c; }
+    static int F() => 0;
+    static void Main()
+    {
+        var c = new C();
+        _ = c + F;
+    }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (11,13): error CS0019: Operator '+' cannot be applied to operands of type 'C' and 'method group'
+                //         _ = c + F;
+                Diagnostic(ErrorCode.ERR_BadBinaryOps, "c + F").WithArguments("+", "C", "method group").WithLocation(11, 13));
+
+            var expectedDiagnostics = new[]
+            {
+                // (11,17): error CS0428: Cannot convert method group 'F' to non-delegate type 'Expression'. Did you intend to invoke the method?
+                //         _ = c + F;
+                Diagnostic(ErrorCode.ERR_MethGrpToNonDel, "F").WithArguments("F", "System.Linq.Expressions.Expression").WithLocation(11, 17)
+            };
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(expectedDiagnostics);
+            comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(expectedDiagnostics);
+        }
+
+        [Fact]
+        public void BinaryOperator_08()
+        {
+            var source =
+@"using System;
+class A
+{
+    public static A operator+(A a, Func<int> f) { Console.WriteLine(""operator+(A a, Func<int> f)""); return a; }
+}
+class B : A
+{
+    public static B operator+(B b, Delegate d) { Console.WriteLine(""operator+(B b, Delegate d)""); return b; }
+    static int F() => 1;
+    static void Main()
+    {
+        var b = new B();
+        _ = b + F;
+        _ = b + (() => 2);
+    }
+}";
+
+            CompileAndVerify(source, parseOptions: TestOptions.Regular9, expectedOutput:
+@"operator+(A a, Func<int> f)
+operator+(A a, Func<int> f)
+");
+
+            // Breaking change from C#9.
```

Comment:
Let's update the breaking change doc along with this and review that we've documented other breaks we've accepted. Could be done in a separate PR, but feels like it would fit well along with this.